### PR TITLE
[Test] Add missing tests to bazel BUILD

### PR DESCRIPTION
**Pull request by jjyao**
## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by jjyao** [comment]
Caught by running 
```
for f in python/ray/tests/test_*.py
do
  f=$(basename -- "$f")
  f="${f%.*}"
  echo "$f"
  ~/bin/bazel cquery "//python/ray/tests:$f"
done
```
and check for `ERROR: no such target`

**Review by mwtian — COMMENTED** [review]
Good find! Has it been verified that `size = "small"` is appropriate for all of them, and is there any test that is intentionally disabled / ran manually / included in a target with a different name? We can find out after CI finishes.

**Comment by jjyao** [comment]
We need a better solution here, several options:

1.  Add a test to make sure each test_*.py file has a corresponding bazel target, similar to what I did above.
2. Auto populate py_test targets using `glob`, but we need a way to figure out `size`, `tags`, `team` that are specific to each test. Maybe a particular folder structure like `python/ray/tests/core/small/test_**.py`? Or the first line of the test file is a comment containing those information?

Also how do we make sure those bazel targets are run by the buildkite?

**Comment by jjyao** [comment]
It's annoying that github doesn't allow me to comment on untouched code. 
```
py_test(
    name = "test_serve_ray_minimal",
    size = "small",
    srcs = SRCS + ["test_serve_ray_minimal.py"],
    tags = ["exclusive", "python_tests", "team:serve"],
    deps = ["//:ray_lib"],
)
```

Will this test be run by buildkite, it has a special `python_tests` tag that's not used by other tests.

**Comment by mwtian** [comment]
Maybe we can add a genrule similar to your command, to make sure each test file has a corresponding target. Manual / disabled tests can be tagged.

**Comment by ericl** [comment]
I like (1)-- seems that can be a script we run in LINT that checks all test files are included in a target.

File: python/ray/tests/BUILD [hunk 1]
```diff
@@ -146,6 +146,21 @@ py_test_module_list(
     "test_list_actors_2.py",
     "test_list_actors_3.py",
     "test_list_actors_4.py",
+    "test_actor_lifetime.py",
+    "test_actor_out_of_order.py",
+    "test_cross_language.py",
+    "test_dashboard.py",
+    "test_environ.py",
+    "test_k8s_cluster_launcher.py",
+    "test_logging.py",
```
Comment (simon-mo) [inline python/ray/tests/BUILD:155]: this is failing atm https://github.com/ray-project/ray/issues/22740

File: python/ray/tests/BUILD [hunk 2]
```diff
@@ -146,6 +146,21 @@ py_test_module_list(
     "test_list_actors_2.py",
     "test_list_actors_3.py",
     "test_list_actors_4.py",
+    "test_actor_lifetime.py",
+    "test_actor_out_of_order.py",
+    "test_cross_language.py",
+    "test_dashboard.py",
+    "test_environ.py",
+    "test_k8s_cluster_launcher.py",
```
Comment (simon-mo) [inline python/ray/tests/BUILD:154]: this is manual

**Review by rkooo567 — APPROVED** [review]

**Comment by rkooo567** [comment]
Wonder if we have a way to avoid this..

**Review by ericl — APPROVED** [review]
LGTM if these tests pass without flaking.

File: rllib/BUILD [hunk 3]
```diff
@@ -2886,3 +2894,28 @@ py_test(
...
+    "utils/tests/test_check_env.py",
+    "utils/tests/test_check_multi_agent.py",
+    "utils/tests/test_errors.py",
+    "utils/tests/test_utils.py",
+  ],
+  size = "large",
+  extra_srcs = [],
+  deps = [],
+  tags = ["manual", "team:ml"],
+)
```
Comment (gjoliver) [inline rllib/BUILD:2921]: add a newline?

File: ci/travis/check-test-run.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        src_file = src_element.attrib["value"][2:].replace(":", "/")
+        src_files.add(src_file)
+
+    missing_bazel_targets = []
+    for f in glob.glob("python/**/tests/test_*.py", recursive=True):
+        if f.startswith("python/build/") or f.startswith(
+            "python/ray/thirdparty_files/"
+        ):
+            continue
+        # TODO(jiaodong) Remove this once experimental module is tested
```
Comment (jjyao) [inline ci/travis/check-test-run.py:26]: @jiaodong FYI.

File: python/ray/tests/BUILD [hunk 5]
```diff
@@ -8,12 +8,28 @@ SRCS = [] + select({
...
 
+py_test_module_list(
+  files = [
+    "test_dashboard.py",
+    "test_logging.py",
+    "test_ray_cluster_with_external_redis.py",
+    "test_k8s_cluster_launcher.py",
+    "test_runtime_env_working_dir_3.py",
+    "test_setup_worker.py",
+  ],
```
Comment (jjyao) [inline python/ray/tests/BUILD:19]: These are currently failing, I'll follow up with owners to fix them.

File: python/ray/tune/BUILD [hunk 6]
```diff
@@ -374,6 +376,17 @@ py_test(
...
+py_test_module_list(
+  files = [
+      "tests/test_trial_executor_inheritance.py",
+      "tests/test_trial_runner_utils.py",
+  ],
+  size = "large",
+  extra_srcs = [],
+  tags = ["exclusive", "manual", "team:ml"],
+  deps = ["//:tune_lib"],
+)
```
Comment (jjyao) [inline python/ray/tune/BUILD:388]: @xwjiang2010 Could you take a look later on to see whether we want to enable them or not?

File: python/ray/util/sgd/BUILD [hunk 7]
```diff
@@ -90,6 +92,16 @@ py_test(
     tags = ["team:ml", "exclusive", "pytorch"],
     deps = [":sgd_lib"],
 )
+
+py_test_module_list(
```
Comment (jjyao) [inline python/ray/util/sgd/BUILD:96]: @amogkam Could you take a look later on to see whether we want to enable them or not?

Comment (jjyao) [reply python/ray/util/sgd/BUILD]: Removed.

File: python/ray/tune/BUILD (hunk 6 above)
Comment (xwjiang2010) [reply python/ray/tune/BUILD:388]: `test_trial_runner_utils` is actually a testing util. Should we rename it to avoid confusion?

Comment (jjyao) [reply python/ray/tune/BUILD]: Renamed to utils_for_test_trail_runner.py

**Review by simon-mo — APPROVED** [review]
LGTM on CI infra side

**Comment by jjyao** [comment]
It's ready to be merged.

**Review by ericl — APPROVED** [review]

**Review by avnishn — APPROVED** [review]

**Pull request merged**